Eevee.Sleep.Bot/Workers/InGameAnnouncementCrawlingWorker.cs

```diff
@@ -0,0 +1,85 @@
+using AngleSharp.Common;
+using Discord.WebSocket;
+using Eevee.Sleep.Bot.Controllers.Mongo.InGameAnnouncement;
+using Eevee.Sleep.Bot.Enums;
+using Eevee.Sleep.Bot.Exceptions;
+using Eevee.Sleep.Bot.Extensions;
+using Eevee.Sleep.Bot.Models.InGameAnnouncement;
+using Eevee.Sleep.Bot.Utils.DiscordMessageMaker;
+using Eevee.Sleep.Bot.Workers.Scrapers.InGameAnnouncement;
+
+namespace Eevee.Sleep.Bot.Workers;
+public class InGameAnnouncementCrawlingWorker(
+    DiscordSocketClient client,
+    ILogger<InGameAnnouncementCrawlingWorker> logger
+) : BackgroundService {
+    private readonly DiscordSocketClient _client = client;
+    private readonly ILogger<InGameAnnouncementCrawlingWorker> _logger = logger;
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+
+    private static readonly TimeSpan CheckInterval = TimeSpan.FromSeconds(60);
+
+    private static async Task<IEnumerable<InGameAnnouncementIndexModel>> GetIndexes() {
+        Dictionary<string, InGameAnnoucementLanguage> BaseUrls = new(){
+            { "https://www.pokemonsleep.net/news", InGameAnnoucementLanguage.JP },
+            { "https://www.pokemonsleep.net/en/news", InGameAnnoucementLanguage.EN },
+            { "https://www.pokemonsleep.net/zh/news", InGameAnnoucementLanguage.ZH },
+        };
+
+        var tasks = BaseUrls.Select(dict => IndexScraper.GetAllPagesAsync(dict.Key, dict.Value)).ToArray();
+        var results = await Task.WhenAll(tasks);
+
+        return results.SelectMany(x => x);
+    }
+
+    private static async Task<IEnumerable<InGameAnnouncementDetailModel>> GetDetails(IEnumerable<InGameAnnouncementIndexModel> indexes) {
+        var detailTasks = indexes
+            .AsParallel()
+            .WithDegreeOfParallelism(5)
+            .Select(DetailScraper.GetAsync);
+    
+        return await Task.WhenAll(detailTasks);
+    }
+
+    private static async Task SaveDetailsAndHistories(IEnumerable<InGameAnnouncementDetailModel> details) {
+        var existedDetails = InGameAnnouncementDetailController.FindAllByIds(details.Select(x => x.AnnouncementId));
+        var existedDetailsById = existedDetails.ToDictionary(x => x.AnnouncementId);
+    
+        var shouldSaveDetail = new List<InGameAnnouncementDetailModel>();
+    
+        foreach (var detail in details) {
+            var detailModel = existedDetailsById.GetOrDefault(detail.AnnouncementId, null);
+            if (detailModel?.ContentHash != detail.ContentHash) {
+                shouldSaveDetail.Add(detail!);
```

Comment:
Maybe check `detail is null` before entering this is? Try to avoid `!` because it assumes that `detail` won't be `null` but any **assumption** is a potential danger of breaking the code

Reply:
Obviously unnecessary `!` so I'm deleting it.